src/components/pages/components/LinkResolver/Resolvers/StudyLogin.tsx

```diff
@@ -108,14 +126,16 @@ const StudyLogin: React.FC<StudyLoginProps> = (props) => {
         dispatch(dialogActions.openDialogWithoutPayload({ type: dialog }));
       }}
       onSubmit={(email, password, rememberMe) => {
+        setLoginInitiated(true);
         dispatch(dialogActions.openLoginDialog(
           {
             type: 'login',
             payload: {
               email: email,
               password: password,
               verificationCode: loginData.verificationCode,
-              rememberMe: rememberMe
+              rememberMe: rememberMe,
+              preventNavigateOnSuccess: (redirectTo !== undefined && redirectTo !== null && redirectTo.length > 0),
```

Comment:
The way `redirectTo` is validated is inconsistent. Here the check is `redirectTo !== undefined && redirectTo !== null && redirectTo.length > 0` but in other places is only `redirectTo !== undefined && redirectTo.length > 0`. 
We might want to centralize this check in a function so that we use it in a consistent way.

Reply:
If we use a centralised check, we also need to use `as string` whenever the value of `redirectTo` is assigned to a value that cannot be undefined, otherwise the compiler complains about it... (It only recognises the check if it's done inline) But yeah, since it used at three places at least, it's probably worth still. The null check is not necessary here anymore, was left in it from when it used the query param directly, which can be null.

Reply:
I cannot close this comment because I don't have write access, but it can be closed after the new commit.